modules/openthread/platform/radio_nrf5.c

```diff
@@ -0,0 +1,1803 @@
+			/**
+			 * Subtract a 32-bit overflow and replace the least significant
+			 * 32 bits with the provided target time.
+			 */
+			result = now_us - UINT32_MAX - 1;
+			result &= ~(uint64_t)UINT32_MAX;
+			result |= target_time_us_wrapped;
+		}
+	}
+
+	__ASSERT_NO_MSG(result <= INT64_MAX / NSEC_PER_USEC);
+	return (int64_t)result * NSEC_PER_USEC;
+}
+
+void platformRadioInit(void)
+{
+	nrf5_data.state = OT_RADIO_STATE_DISABLED;
+
+	/* Get the default tx output power from Kconfig */
+	nrf5_data.tx_power = CONFIG_OPENTHREAD_DEFAULT_TX_POWER;
+
+	for (size_t i = 0; i < CHANNEL_COUNT; i++) {
+		nrf5_data.max_tx_power_table[i] = OT_RADIO_POWER_INVALID;
+	}
+
+	nrf5_data.rx_on_when_idle = true;
+
+	nrf5_get_eui64(nrf5_data.mac);
+
+	k_fifo_init(&nrf5_data.rx.fifo);
+
+	nrf5_data.tx.frame.mPsdu = PSDU_DATA(nrf5_data.tx.psdu);
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
+	nrf5_data.tx.frame.mInfo.mTxInfo.mIeInfo = &nrf5_data.tx.ie_info;
+#endif
+
+	k_sem_init(&nrf5_data.rssi_wait, 0, 1);
+
+	nrf_802154_init();
+
+	nrf5_data.capabilities = nrf5_get_caps();
+}
+
+static void openthread_handle_received_frame(otInstance *instance, struct nrf5_rx_frame *rx_frame)
+{
+	otRadioFrame recv_frame;
+	uint8_t *psdu;
+
+	ARG_UNUSED(instance);
+
+	__ASSERT_NO_MSG(rx_frame->psdu != NULL);
+
+	memset(&recv_frame, 0, sizeof(otRadioFrame));
+
+	recv_frame.mPsdu = PSDU_DATA(rx_frame->psdu);
+	/* Length inc. CRC. */
+	recv_frame.mLength = PSDU_LENGTH(rx_frame->psdu);
+	recv_frame.mChannel = nrf5_data.channel;
+	recv_frame.mInfo.mRxInfo.mLqi = rx_frame->lqi;
+	recv_frame.mInfo.mRxInfo.mRssi = rx_frame->rssi;
+	recv_frame.mInfo.mRxInfo.mAckedWithFramePending = rx_frame->ack_fpb;
+	recv_frame.mInfo.mRxInfo.mTimestamp = rx_frame->time;
+	recv_frame.mInfo.mRxInfo.mAckedWithSecEnhAck = rx_frame->ack_seb;
+
+	LOG_DBG("RX %p len: %u, ch: %u, rssi: %d", (void *)recv_frame.mPsdu, recv_frame.mLength,
+		recv_frame.mChannel, recv_frame.mInfo.mRxInfo.mRssi);
+
+	if (IS_ENABLED(CONFIG_OPENTHREAD_DIAG) && otPlatDiagModeGet()) {
+		otPlatDiagRadioReceiveDone(instance, &recv_frame, OT_ERROR_NONE);
+	} else {
+		otPlatRadioReceiveDone(instance, &recv_frame, OT_ERROR_NONE);
+	}
+
+	psdu = rx_frame->psdu;
+	rx_frame->psdu = NULL;
+	nrf_802154_buffer_free_raw(psdu);
+}
+
+static void energy_detected(int16_t max_ed)
+{
+	nrf5_data.energy_detection.value = max_ed;
+	set_pending_event(PENDING_EVENT_DETECT_ENERGY_DONE);
+}
+
+static bool nrf5_tx(otRadioFrame *frame, uint8_t *payload, bool cca)
+{
+	nrf_802154_transmit_metadata_t metadata = {
+		.frame_props =
+			{
+				.is_secured = frame->mInfo.mTxInfo.mIsSecurityProcessed,
+				.dynamic_data_is_set = frame->mInfo.mTxInfo.mIsHeaderUpdated,
+			},
+		.cca = cca,
+		.tx_power =
+			{
+				.use_metadata_value = true,
+				.power = get_transmit_power_for_channel(frame->mChannel),
+			},
+	};
+
+	return nrf_802154_transmit_raw(payload, &metadata);
+}
+
+#if NRF_802154_CSMA_CA_ENABLED
+static bool nrf5_tx_csma_ca(otRadioFrame *frame, uint8_t *payload)
```

Comment:
I can see that in this file there is a bunch of methods that use input ptrs without nullptr checks. Are we sure it's safe? Are there check anywhere before, so we don't need to repeat it here and there?